packages/@lwc/engine-core/src/framework/hydration.ts

```diff
@@ -751,7 +757,7 @@ function areCompatibleNodes(client: Node, ssr: Node, vnode: VNode, renderer: Ren
                     client,
                     'tagName'
                 ).toLowerCase()}" but found "${getProperty(ssr, 'tagName').toLowerCase()}".`,
-                vnode.owner
+                owner
             );
         }
```

Comment:
Looking at the code coverage report (click into `run-karma-tests`, then download the combined coverage report), basically all of the above code in `areCompatibleStaticNodes` is not actually hit.

![Screenshot 2024-03-21 at 4 28 25 PM](https://github.com/salesforce/lwc/assets/283842/32c072f9-e984-435d-864f-ac35134559e6)

Reply:
Hrrm, I think the first two will never be hit for static nodes because of the way we're parsing them, the top level element will always need to be an [`Element`](https://github.com/salesforce/lwc/blob/master/packages/%40lwc/engine-core/src/framework/hydration.ts#L227).

Maybe we should remove it, the `tagName` check is interesting, I'm not sure how we can trigger that to happen in our current tests.

Reply:
I wonder if this function was intended to be called recursively through the entire subtree instead of just at the root node?

Otherwise, the comment and text checks will never be reached.

Reply:
Let's look into this in a follow-up PR. You're right that the top-level should always be an Element.